pkg/Cpanel/Security/Advisor/Assessors/ClamAV.pm

```diff
@@ -82,22 +84,43 @@ sub _check_clamav {
                 'text'       => $self->_lh->maketext('ClamAV freshclam binary is not installed.'),
                 'suggestion' => $self->_lh->maketext(
                     'Install ClamAV within "[output,url,_1,Manage Plugins,_2,_3]".',
-                    $self->base_path('scripts2/getthemes?modules=1'), 'target', '_blank',
+                    $self->base_path($install_clamav), 'target', '_blank',
                 ),
             }
         );
     }
     else {
         $self->_get_clam_version();
+
+        my @bad_clams;
+
+        push( @bad_clams, 'clamscan' )  if !defined $self->{clamav}{clamscan}{version};
+        push( @bad_clams, 'freshclam' ) if !defined $self->{clamav}{freshclam}{version};
+
+        if (@bad_clams) {
+
+            $security_advisor_obj->add_advice(
+                {
+                    'key'        => 'ClamAV_failed_to_get_version',
+                    'type'       => $Cpanel::Security::Advisor::ADVISE_BAD,
+                    'text'       => $self->_lh->maketext( 'Failed to get version for [list_and,_1].', \@bad_clams ),
+                    'suggestion' => $self->_lh->maketext(
+                        'clamscan version: [_1]<br/>freshclam version: [_2]<br/><br/>Install ClamAV within "[output,url,_3,Manage Plugins,_4,_5]".', $self->{clamav}{clamscan}{version_str}, $self->{clamav}{freshclam}{version_str}, $self->base_path($install_clamav), 'target', '_blank',
```

Comment:
We may want to have a different suggestion on this item.  This issue only appears when we fail to run extract the version information.  My worry is that people get into a loop, where the suggestion is to Install ClamAV, so they do, but the problem doesn't get resolved and they are asked to install it again, ad nauseam.

If we know why this issue comes up, and that a re-install would clearly fix it, then I'm more open to leaving this as is.  Though, it might be better to say "Re-install" instead of "Install".

Reply:
This is the "catch-all" failure, so it is hard to give very specific advice here. The only way I have triggered it was by intentionally corrupting the install. So I would not be against rewording to the advice to "Reinstall".

I would not be too worried about people getting into a loop. If they do get this message again after reinstall then there is good reason to investigate further and possibly open a ticket. I do not believe a person would just continue to reinstall, they are after all a thinking human being and not a computer constrained by its limited logic.